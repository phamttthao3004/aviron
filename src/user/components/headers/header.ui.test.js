import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import HeaderUI from './header.ui';
import CONSTANTS from '../../scripts/apps/constants';

describe('Header UI', () => {
    it('Render Header UI component with currentPage=undefined', () => {
        const props = { currentPage: undefined, clickMenu: undefined, openCart: undefined, setOpenCart: undefined, setBack: undefined };
        const com = shallow(<HeaderUI {...props} />);
        expect(shallowToJson(com)).toMatchSnapshot();
    });

    it('Render Header UI component with currentPage=CONSTANTS.SIDEBAR_PAGEID.PROFILE', () => {
        const props = { currentPage: CONSTANTS.SIDEBAR_PAGEID.PROFILE, clickMenu: undefined, openCart: undefined, setOpenCart: undefined, setBack: undefined };
        const com = shallow(<HeaderUI {...props} />);
        expect(shallowToJson(com)).toMatchSnapshot();
    });

    it('Render Header UI component with currentPage=CONSTANTS.SIDEBAR_PAGEID.EQUIPMENT', () => {
        const props = { currentPage: CONSTANTS.SIDEBAR_PAGEID.EQUIPMENT, clickMenu: undefined, openCart: undefined, setOpenCart: undefined, setBack: undefined };
        const com = shallow(<HeaderUI {...props} />);
        expect(shallowToJson(com)).toMatchSnapshot();
    });
});
