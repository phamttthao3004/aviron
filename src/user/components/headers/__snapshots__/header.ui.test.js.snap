// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header UI Render Header UI component with currentPage=CONSTANTS.SIDEBAR_PAGEID.EQUIPMENT 1`] = `
<div
  id="header"
>
  <Top
    clickMenu={[Function]}
    currentPage={200}
  />
  <NavUI
    clickMenu={[Function]}
    currentPage={200}
    openCart={false}
    setBack={[Function]}
    setOpenCart={[Function]}
  />
</div>
`;

exports[`Header UI Render Header UI component with currentPage=CONSTANTS.SIDEBAR_PAGEID.PROFILE 1`] = `
<div
  id="header"
>
  <Top
    clickMenu={[Function]}
    currentPage={100}
  />
  <NavUI
    clickMenu={[Function]}
    currentPage={100}
    openCart={false}
    setBack={[Function]}
    setOpenCart={[Function]}
  />
</div>
`;

exports[`Header UI Render Header UI component with currentPage=undefined 1`] = `
<div
  id="header"
>
  <Top
    clickMenu={[Function]}
    currentPage={-1}
  />
  <NavUI
    clickMenu={[Function]}
    currentPage={-1}
    openCart={false}
    setBack={[Function]}
    setOpenCart={[Function]}
  />
</div>
`;
